units/host_drawdown.go

```diff
@@ -122,6 +122,11 @@ func (j *hostDrawdownJob) checkAndTerminateHost(ctx context.Context, h *host.Hos
 		return err
 	}
 
+	// don't drawdown hosts that are currently in the middle of tearing down a task group
+	if h.IsTearingDown {
```

Comment:
We set isTearingDown before we return `nextTaskResponse.ShouldTeardownGroup = true` along with a nil task to the agent. If it reached the line that set isTearingDown to true, it's because it determined that a task group should be torn down and decided not to assign a new task. This is done at the same time as assignNextAvailableTask setting next task to nil, not on the next iteration.  

However, to avoid the concern of it potentially getting stuck in IsTearingDown forever I updated this check so it now takes into consideration the time it started.